refAllDecls does not handle pub instances of structs

### Zig Version

0.10.0-dev.760+1a84c23d6

### Steps to Reproduce

Test the below file.

```zig
const std = @import("std");

pub const SInstance: S = .{};

pub const S = struct {
    a: usize = 0,
    b: bool = false,
};

comptime {
    std.testing.refAllDecls(@This());
}
```

### Expected Behavior

No compile error.

### Actual Behavior

```
zig test src/main.zig                                                                                                                                                 237ms  Thu 17 Feb 2022 00:54:08 GMT
/home/lee/src/github.com/leecannon/zig/lib/std/testing.zig:470:26: error: expected type 'type', found 'S'
            _ = @hasDecl(@field(T, decl.name), "foo");
                         ^
./src/main.zig:11:28: note: called from here
    std.testing.refAllDecls(@This());
                           ^
./src/main.zig:10:10: note: called from here
comptime {
         ^
```

--- a/lib/std/testing.zig
+++ b/lib/std/testing.zig
@@ -466,13 +466,6 @@ test {
 pub fn refAllDecls(comptime T: type) void {
     if (!builtin.is_test) return;
     inline for (comptime std.meta.declarations(T)) |decl| {
-        if (decl.is_pub) {
-            const decl_value = @field(T, decl.name);
-            const type_of_decl = @TypeOf(decl_value);
-            if (type_of_decl == type and @typeInfo(type_of_decl) == .Struct) {
-                _ = @hasDecl(decl_value, "foo");
-            }
-        }
-        _ = decl;
+        if (decl.is_pub) _ = @field(T, decl.name);
     }
 }

--- a/lib/std/testing.zig
+++ b/lib/std/testing.zig
@@ -466,8 +466,13 @@ test {
 pub fn refAllDecls(comptime T: type) void {
     if (!builtin.is_test) return;
     inline for (comptime std.meta.declarations(T)) |decl| {
-        if (decl.is_pub and @typeInfo(@TypeOf(@field(T, decl.name))) == .Struct)
-            _ = @hasDecl(@field(T, decl.name), "foo");
+        if (decl.is_pub) {
+            const decl_value = @field(T, decl.name);
+            const type_of_decl = @TypeOf(decl_value);
+            if (type_of_decl == type and @typeInfo(type_of_decl) == .Struct) {
+                _ = @hasDecl(decl_value, "foo");
+            }
+        }
         _ = decl;
     }
 }

